src/components/PaymentCard/PaymentCard.js

```diff
@@ -3,88 +3,123 @@ import PropTypes from 'prop-types';
 import { useTranslation } from 'react-i18next';
 import SkeletonWrapper from 'components/SkeletonWrapper';
 import { POPUP_TYPES } from 'redux/innerPopupReducer';
-import { CardTypesIcons } from './Payment.const';
+import Card from 'components/Card';
+import eventDispatcher, {
+  MSSDK_EDIT_PAYMENT_BUTTON_CLICKED
+} from 'util/eventDispatcher';
+import { CardTypes } from './Payment.const';
 import {
   CardStyled,
-  CardWrapStyled,
   CardTypeStyled,
   CardNumberStyled,
   CardExpirationStyled,
   CardExpirationLabel,
   CardExpirationDateStyled,
   CardEditStyled,
-  MethodNameStyled
+  CardInfoStyled,
+  CardDetailsStyled,
+  CardDetailsNameStyled,
+  CardDetailsNameWrapStyled,
+  CardInfoWrapStyled,
+  HolderNameStyled
 } from './PaymentCardStyled';
 
-const INAPPS = ['android', 'apple', 'amazon', 'roku'];
+const PaymentCardSkeleton = () => (
+  <CardStyled>
+    <CardInfoWrapStyled>
+      <CardInfoStyled>
+        <CardTypeStyled>
+          <SkeletonWrapper height={16} width={28} />
+        </CardTypeStyled>
+        <CardDetailsStyled>
+          <CardDetailsNameWrapStyled>
+            <CardDetailsNameStyled>
+              <SkeletonWrapper width={140} />
+            </CardDetailsNameStyled>
+          </CardDetailsNameWrapStyled>
+          <CardExpirationStyled>
+            <SkeletonWrapper width={100} />
+          </CardExpirationStyled>
+        </CardDetailsStyled>
+      </CardInfoStyled>
+      <SkeletonWrapper height={43} width={170} />
+    </CardInfoWrapStyled>
+  </CardStyled>
+);
+
+const getSpecificPaymentMethod = ({ variant }, paymentMethod) => {
+  if (paymentMethod === 'card') return variant;
+  return paymentMethod;
+};
+
 const PaymentCard = ({ isDataLoaded, details, showInnerPopup }) => {
   const { t } = useTranslation();
   const { paymentMethodSpecificParams, paymentMethod } = details;
-  const LogoComponent =
-    paymentMethodSpecificParams &&
-    CardTypesIcons[paymentMethodSpecificParams.variant]
-      ? CardTypesIcons[paymentMethodSpecificParams.variant]
-      : CardTypesIcons[details.paymentMethod] || null;
+
+  const { icon: LogoComponent, title: methodTitle } = isDataLoaded
+    ? CardTypes[
+        getSpecificPaymentMethod(paymentMethodSpecificParams, paymentMethod)
+      ]
+    : { icon: null, title: '' };
```

Comment:
if we would put getSpecificPaymentMethod into PaymentCard component, we won't need to pass params (like in line 61)